.github/workflows/test_asterinas.yml

```diff
@@ -24,6 +24,20 @@ jobs:
         id: check
         run: make check
 
+  compilation:
+    if: github.event_name == 'push' || github.event_name == 'pull_request'
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    container: asterinas/asterinas:0.9.4
+    steps:
+      - run: echo "Running in asterinas/asterinas:0.9.4"
+
+      - uses: actions/checkout@v4
+
+      - name: Compilation
+        id: compilation
+        run: make build FEATURES=all
```

Comment:
Add a comment here to explain why we have a compilation test even when the subsequent unit and integration tests already include compilation.

```yml
      # The compilation test builds the project with all features enabled.
      # In contrast, subsequent tests may choose to enable select features.
      - name: Compilation
```